aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -169,49 +240,19 @@ module aptos_framework::transaction_validation {
                 txn_sequence_number == 0,
                 error::invalid_argument(PROLOGUE_ESEQUENCE_NUMBER_TOO_NEW)
             );
-
-            if (!features::transaction_simulation_enhancement_enabled() ||
-                    !skip_auth_key_check(is_simulation, &txn_authentication_key)) {
-                if (option::is_some(&txn_authentication_key)) {
-                    assert!(
-                        txn_authentication_key == option::some(bcs::to_bytes(&transaction_sender)),
-                        error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
-                    );
-                } else {
-                    assert!(
-                        allow_missing_txn_authentication_key(transaction_sender),
-                        error::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY)
-                    );
-                }
-            };
         };
+    }
 
-        let max_transaction_fee = txn_gas_price * txn_max_gas_units;
-
-        if (!features::transaction_simulation_enhancement_enabled() || !skip_gas_payment(
-            is_simulation,
-            gas_payer_address
-        )) {
-            assert!(
-                permissioned_signer::check_permission_capacity_above(
-                    gas_payer,
-                    (max_transaction_fee as u256),
-                    GasPermission {}
-                ),
-                error::permission_denied(PROLOGUE_PERMISSIONED_GAS_LIMIT_INSUFFICIENT)
-            );
-            if (features::operations_default_to_fa_apt_store_enabled()) {
-                assert!(
-                    aptos_account::is_fungible_balance_at_least(gas_payer_address, max_transaction_fee),
-                    error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
-                );
-            } else {
-                assert!(
-                    coin::is_balance_at_least<AptosCoin>(gas_payer_address, max_transaction_fee),
-                    error::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT)
-                );
-            }
-        }
+    fun check_for_replay_protection_orderless_txn(
+        sender: address,
+        nonce: u64,
+        txn_expiration_time: u64,
+    ) {
+        assert!(
+            txn_expiration_time <= timestamp::now_seconds() + MAX_EXPIRATION_TIME_FOR_ORDERLESS_TXNS,
```

Comment:
We do not check that `txn_expiration_time` is too old?

Reply:
Found it: can you add a comment here saying that prologue checks `timestamp::now_seconds() < txn_expiration_time` at the start?

Reply:
Added the comment.